lisa/schema.py

```diff
@@ -504,6 +504,107 @@ def _generate_min_capability(self, capability: Any) -> Any:
         return min_value
 
 
+class NetworkDataPath(str, Enum):
+    Synthetic = "Synthetic"
+    Sriov = "Sriov"
+
+
+_network_data_path_priority: List[NetworkDataPath] = [
+    NetworkDataPath.Sriov,
+    NetworkDataPath.Synthetic,
+]
+
+
+def _decode_network_data_path_type(data: Any) -> Any:
```

Comment:
This logic can be shared by using generic type. The name can be `decode_set_space_by_type`.

Reply:
You can check from the implementation of factory.